source/features/prevent-pr-commit-link-loss.tsx

```diff
@@ -1,16 +1,33 @@
-import delegate from 'delegate-it';
 import * as pageDetect from 'github-url-detection';
+import AlertIcon from 'octicon/alert.svg';
+import delegate from 'delegate-it';
+import React from 'dom-chef';
+import select from 'select-dom';
 
 import features from '../libs/features';
-import {preventPrCommitLinkBreak} from '../libs/utils';
+import {containsPrCommitLink, preventPrCommitLinkBreak} from '../libs/utils';
 
 function init(): void {
-	delegate(document, 'form#new_issue, form.js-new-comment-form', 'submit', updateTextArea);
+	delegate(document, 'form#new_issue textarea, form.js-new-comment-form textarea', 'input', handleTextAreaChange);
 }
 
-function updateTextArea(event: delegate.Event): void {
-	const field = event.delegateTarget.querySelector('textarea')!;
-	field.value = preventPrCommitLinkBreak(field.value);
+function handleTextAreaChange(event: delegate.Event): void {
+	const field = (event.delegateTarget as HTMLInputElement);
+	const fieldValue = field.value;
+
+	if(containsPrCommitLink(fieldValue) && !select.exists('#fix-pr-commit-links')) {
+		const closestForm = field.closest('form')!;
+		select('.form-actions', closestForm)?.prepend(
+			<div className="flash flash-warn mb-2">
+			  <AlertIcon /> Your PR Commit link may be <a href="https://github.com/sindresorhus/refined-github/issues/2327">misinterpreted by GitHub.</a>
+			  <button type="button" className="btn btn-sm primary flash-action" id="fix-pr-commit-links">Fix link</button>
```

Comment:
Mixed whitespace at the beginning of the lines.

Also it's best to use classes starting with `rgh-` instead of IDs
```suggestion
				<button type="button" className="btn btn-sm primary flash-action rgh-fix-pr-commit-links">Fix link</button>
```